app/api/csv/typeFormatters.ts

```diff
@@ -0,0 +1,43 @@
+import moment from 'moment';
+
+export const formatDate = (timestamp: number, format: string = 'YYYY-MM-DD') =>
+  moment.unix(timestamp).format(format);
+
+const formatFile = (fileName: string) => `/files/${fileName}`;
+const formatAttachment = (fileName: string, entityId: string) =>
+  `/api/attachments/download?_id=${entityId}&file=${fileName}`;
+
+const formatters: {
+  [key: string]: (field: any, options?: any) => string;
+} = {
+  select: (field: any) => (field && field[0] ? field[0].label : ''),
+  multiselect: (field: any) => field?.map((item: any) => formatters.select([item])).join('|') || '',
```

Comment:
I was not doing the checks at first, but I got errors because I got rows in the results for which the template indicated there should be a field, while it was actually not set in the row. The OR operand is related to the optional chaining of map. It will make the whole instruction return an undefined value and the OR defaults it to the empty string.

Anyways, this is clearly not the place to do those checks as the formatters should only transform an existing field. I'll remove them from here and try to reproduce the failing scenarios.